### ISSUE TITLE
127.0.0.1 is not properly escaped in agent's run.sh

### ISSUE THREAD

**Issue by cytadelle** [rancher/rancher#9176]
**Rancher versions:**
rancher/agent:

Hello,
We have found a bug which is causing "ERROR: http://xxxxx/v1 is not accessible"

My CATTLE_URL was: http://host.com/v1/scripts/B1A21CAB5127F00041F5:1483142400000:bpVjLfawHViOVh2QSPbfSPDmMY

After "bad setup":
http://host.com/v1/scripts/B1A21CAB510.0.10.10F5:1483142400000:bpVjLfawHViOVh2QSPbfSPDmMY

Unextected sed replacement:
127F00041 was replaced with the gateway ip "10.0.10.10"

https://github.com/rancher/rancher/blob/master/agent/run.sh => line 416

CATTLE_URL=$(echo $CATTLE_URL | sed -e 's/127.0.0.1/'$gateway'/' -e 's/localhost/'$gateway'/')

I think 127.0.0.1 is seen as a regular expression by sed... So you should escape the . by \\. and replace with 127\\.0\\.0\\.1
You could also use something like ^http(s?)://127\\.0\\.0\\.1

Regards
Arnaud

**Comment by superseb** [comment]
Thanks for reporting, this actually happens in https://github.com/rancher/rancher/blob/master/agent/run.sh#L414 I think. Created a PR.

**Comment by cjellick** [comment]
available in master (not in rc2).
There's a small chance that in some hacked-up setup, this could break someone, so it might be worth calling out in the release notes I think

**Comment by sangeethah** [comment]
@cjellick Wanted to confirm that this fix available in v1.6 master ?

**Comment by cjellick** [comment]
ah, yes @sangeethah. available in v1.6-development builds

**Comment by soumyalj** [comment]
Tested with master. Set the Host Registration URL to each of the following and added a host
a) http://127.0.0.1:8080
b) https://127.0.0.1:8080
c) http://localhost:8080
d) https://localhost:8080
Host addition failed and the following error was seen on rancher-agent-bootstrap container as expected:
```
root@soumyamasterhostadd1:~# sudo docker ps -a
CONTAINER ID        IMAGE                      COMMAND                  CREATED             STATUS                      PORTS               NAMES
b050ca22dd60        rancher/agent:v1.2.6-rc1   "/run.sh http://local"   12 minutes ago      Exited (1) 12 minutes ago                       rancher-agent-bootstrap
root@soumyamasterhostadd1:~# sudo docker logs -f b050ca22dd60

ERROR: CATTLE_URL cannot contain localhost or 127.0.0.1, please check the Host Registration URL.
```
